debug_utils/sierra-emu/tests/corelib.rs

```
@@ -0,0 +1,226 @@
+use cairo_lang_starknet::starknet_plugin_suite;
+use cairo_lang_test_plugin::{
+    compile_test_prepared_db,
+    test_config::{PanicExpectation, TestExpectation},
+    test_plugin_suite, TestCompilation, TestsCompilationConfig,
+};
+use common::value_to_felt;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use sierra_emu::{run_program, EntryPoint, ProgramTrace, Value};
+use tracing::info;
+use tracing_test::traced_test;
+
+mod common;
+
+#[traced_test]
+#[test]
+#[ignore = "sierra-emu is not fully implemented yet"]
+fn test_corelib() {
+    let compiler_path = Path::new(&std::env::var("CARGO_MANIFEST_DIR").unwrap()).join("corelib");
+
+    check_compiler_path(false, &compiler_path).expect("Couldn't the corelib in the given path");
+
+    let db = &mut {
+        let mut b = RootDatabase::builder();
+        b.detect_corelib();
+        b.with_cfg(CfgSet::from_iter([Cfg::name("test")]));
+        b.with_default_plugin_suite(test_plugin_suite());
+        b.with_default_plugin_suite(starknet_plugin_suite());
+
+        b.build().unwrap()
+    };
+
+    let main_crate_ids = setup_project(db, &compiler_path).unwrap();
+
+    let db = db.snapshot();
+    let test_crate_ids = main_crate_ids.clone();
+    let test_config = TestsCompilationConfig {
+        starknet: false,
+        add_statements_functions: true,
+        add_statements_code_locations: true,
+        contract_declarations: None,
+        contract_crate_ids: None,
+        executable_crate_ids: None,
+    };
+
+    let diag_reporter = DiagnosticsReporter::stderr().with_crates(&main_crate_ids);
+
+    let filtered_tests = vec![
+        "core::test::dict_test::test_array_from_squash_dict",
+        "core::test::hash_test::test_blake2s",
+        "core::test::testing_test::test_get_unspent_gas",
+    ];
+
+    let compiled = compile_tests(
+        &db,
+        test_config,
+        test_crate_ids,
+        diag_reporter,
+        Some(&filtered_tests),
+    );
+
+    run_tests(compiled);
```

Is there a way to print a report like Cairo Native does? That way we can clearly see which tests are failing and which are not.

---

The cairo native report prints a summary report at the end:
```
test result: ok. 613 passed; 0 failed; 13 ignored; 0 filtered out;
```
We have the same for the sierra emu? I can't find it.

I think another problem is that the tests panic. This implies that the output of some tests is not shown: https://github.com/lambdaclass/cairo_native/actions/runs/14651275989/job/41117426171#step:7:42. Maybe we could catch the panic so that we can clearly see which tests are missing.

---

Well, I've tried to catch panics. There a function `catch_unwind` which retrieved the error message from the panic. The problem is that many panics are todos with no message and even if the had it does not tell where it was originated. It would be better the make a refactor and return errors instead of panicking. I'll leave the code since it helps getting the test which have panicked, but some of them won't tell the actual error.